src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c

```diff
@@ -75,7 +75,27 @@ static HYD_status list_to_nodes(char *str)
 
     string = MPL_strdup(str);
 
-    /* compile regex patterns for nodelist matching */
+    /* compile regex patterns for nodelist matching:
+     *
+     * From RFC952:
+     * 1. A "name" (Net, Host, Gateway, or Domain name) is a text string up
+     * to 24 characters drawn from the alphabet (A-Z), digits (0-9), minus
+     * sign (-), and period (.).  Note that periods are only allowed when
+     * they serve to delimit components of "domain style names". (See
+     * RFC-921, "Domain Name System Implementation Schedule", for
+     * background).  No blank or space characters are permitted as part of a
+     * name. No distinction is made between upper and lower case.  The first
+     * character must be an alpha character.  The last character must not be
+     * a minus sign or period.  A host which serves as a GATEWAY should have
+     * "-GATEWAY" or "-GW" as part of its name.  Hosts which do not serve as
+     * Internet gateways should not use "-GATEWAY" and "-GW" as part of
+     * their names. A host which is a TAC should have "-TAC" as the last
+     * part of its host name, if it is a DoD host.  Single character names
+     * or nicknames are not allowed.
+     *
+     * see https://tools.ietf.org/html/rfc952 and section 2.1 of
+     * https://tools.ietf.org/html/rfc1123 for more details on hostnames
+     * format */
 
```

Comment:
I would take out the "GATEWAY" and "TAC" part as they are not relevant to your code; but it's alright if you feel lazy by now :)

Reply:
Let me double check something and I will also fix that